pkg/resource/resource.go

```diff
@@ -116,3 +125,72 @@ func (r *BaseResource) ID() string {
 
 	return r.id
 }
+
+func (r *BaseResource) Logs() []logger.Entry {
+	return r.logs
+}
+
+// Critical forwards to the logger and adds to the resource logs used for critical errors that interupt
+// the scanner flow.
+func (r *BaseResource) Critical(code logger.ErrorCode, msg string, args ...any) {
+	resourceMsg := fmt.Sprintf("resource_id=%s ", r.id)
+	if entry := logger.Error(resourceMsg+msg, args...); entry != nil {
+		entry.Message = fmt.Errorf(msg, args...).Error()
+		entry.Code = code.String()
+		entry.Severity = "CRITICAL"
+		r.logs = append(r.logs, *entry)
```

Comment:
Now they are tied to resource ids in the stderr it is easier. But it would also be pretty useful to know of any errors that caused my scan to not work without having to tie stdout and stderr together myself. 

Use case i see here is someone just consuming the stdout json objects.

Reply:
Fair :ok_hand: Resolving :D